cli/command/defaultcontextstore.go

```diff
@@ -0,0 +1,182 @@
+}
+
+// CreateOrUpdateContext is not allowed for the default context and fails
+func (s *ContextStoreWithDefault) CreateOrUpdateContext(meta store.ContextMetadata) error {
+	if meta.Name == DefaultContextName {
+		return errors.New("default context and cannot be created nor updated")
+	}
+	return s.Store.CreateOrUpdateContext(meta)
+}
+
+// RemoveContext is not allowed for the default context and fails
+func (s *ContextStoreWithDefault) RemoveContext(name string) error {
+	if name == DefaultContextName {
+		return errors.New("default context and cannot be removed")
+	}
+	return s.Store.RemoveContext(name)
+}
+
+// GetContextMetadata implements store.Store's GetContextMetadata
+func (s *ContextStoreWithDefault) GetContextMetadata(name string) (store.ContextMetadata, error) {
+	if name == DefaultContextName {
+		defaultContext, err := s.Resolver()
+		if err != nil {
+			return store.ContextMetadata{}, err
+		}
+		return defaultContext.Meta, nil
+	}
+	return s.Store.GetContextMetadata(name)
+}
+
+// ResetContextTLSMaterial is not implemented for default context and fails
+func (s *ContextStoreWithDefault) ResetContextTLSMaterial(name string, data *store.ContextTLSData) error {
+	if name == DefaultContextName {
+		return errors.New("The default context store does not support ResetContextTLSMaterial")
+	}
+	return s.Store.ResetContextTLSMaterial(name, data)
+}
+
+// ResetContextEndpointTLSMaterial is not implemented for default context and fails
+func (s *ContextStoreWithDefault) ResetContextEndpointTLSMaterial(contextName string, endpointName string, data *store.EndpointTLSData) error {
+	if contextName == DefaultContextName {
+		return errors.New("The default context store does not support ResetContextEndpointTLSMaterial")
+	}
+	return s.Store.ResetContextEndpointTLSMaterial(contextName, endpointName, data)
+}
+
+// ListContextTLSFiles implements store.Store's ListContextTLSFiles
+func (s *ContextStoreWithDefault) ListContextTLSFiles(name string) (map[string]store.EndpointFiles, error) {
+	if name == DefaultContextName {
+		defaultContext, err := s.Resolver()
+		if err != nil {
+			return nil, err
+		}
+		tlsfiles := make(map[string]store.EndpointFiles)
+		for epName, epTLSData := range defaultContext.TLS.Endpoints {
+			var files store.EndpointFiles
+			for filename := range epTLSData.Files {
+				files = append(files, filename)
+			}
+			tlsfiles[epName] = files
+		}
+		return tlsfiles, nil
+	}
+	return s.Store.ListContextTLSFiles(name)
+}
+
+// GetContextTLSData implements store.Store's GetContextTLSData
+func (s *ContextStoreWithDefault) GetContextTLSData(contextName, endpointName, fileName string) ([]byte, error) {
+	if contextName == DefaultContextName {
+		defaultContext, err := s.Resolver()
+		if err != nil {
+			return nil, err
+		}
+		if defaultContext.TLS.Endpoints[endpointName].Files[fileName] == nil {
+			return nil, errors.Errorf("tls data for %s/%s/%s does not exist", DefaultContextName, endpointName, fileName)
```

Comment:
I think there is a typed error interface for not-found cases in the store package. It should return an error satisfying this interface

Reply:
So just checked, there is no error interface but a specific error type (see here: https://github.com/docker/cli/blob/master/cli/context/store/store.go#L324). I suggest you introduce a tlsDataNotExistError interface in store package, and modify IsErrTLSDataDoesNotExist to check this interface, and makes this code return an error satisfying this interface.